wordpress-mobile/WordPress-iOS
# QR Login: Adds QR Code Scanning

| 📚 Project: #18594 | 🔗  Depends on: https://github.com/wordpress-mobile/WordPress-iOS/pull/18715/ |
|:---:|:---:|

⚠️ 📱 **Note: This PR requires testing on a real device**. 📱 ⚠️

### Description

- Presents the QR code scanning from the Me Menu
- Adds the design for the camera overlay view with some subtle animations
- Provides some basic QR login link validation
- Pushes to the verify auth view (to be done) when a valid QR code is scanned
- If the scanning view is being opened and the app does not have access to the camera:
    - The user will be prompted to enable camera access
    - If access was previously declined, the user will see a prompt to re-enable it via the settings

### Demo

https://user-images.githubusercontent.com/793774/170771626-5ba3fc82-b2a6-47bd-bc14-f7e30d417083.MP4

### To test:
⚠️ 📱 **Note: This PR requires testing on a real device**. 📱 ⚠️

#### Testing camera permissions
1. Uninstall the WordPress app from your device
    2. This is the only way to completely reset the camera permissions 
2. Launch the app via Xcode, and follow the login process
3. Tap on the My Site tab
4. Then tap on your Profile icon in the top corner
5. Tap on the 'Scan Login Code' option
6. 👁️ You should see a prompt asking you to enable camera permissions
7. Tap Don't Allow
8. 👁️ You should see another alert informing you that camera access is required
9. Tap the Cancel button
10. 👁️ The alert should dismiss without proceeding any further
11. Tap the Scan Login Code item again
12. Tap Open Settings
13. You should be brought the Settings.app for the app
14. Enable Camera access and go back to the app
15. Tap the Scan Login Code again
16. 👁️ You should see the camera scanning view

#### Testing QR scanning
1. Bring the camera scanning view up in the app using the previous steps
2. Tap the X button in the top left corner
3. 👁️ The flow dismisses and brings you back to the me menu
4. Tap the Scan Login Code item again
5. Attempt to scan a QR code that is not provided via the QR Login
    - You can scan QR Code examples here: https://en.wikipedia.org/wiki/QR_code
6. 👁️ The QR codes should NOT be scanned and nothing will happen
7. Scan the QR code [from this image](https://user-images.githubusercontent.com/115071/167772947-38a9c302-8edd-4265-8022-0cc695c92ec1.png)
    8. This is a valid (but expired) login code provided by the QR login process
8. 👁️ The code should be scanned and you are brought to a black screen

### Regression Notes
1. Potential unintended areas of impact
None
18. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing
19. What automated tests I added (or what prevented me from doing so)
None

PR submission checklist:
